test_conformance/subgroups/subhelpers.cpp

```diff
@@ -0,0 +1,62 @@
+//
+// Copyright (c) 2022 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "subhelpers.h"
+
+// Define operator<< for cl_ types, accessing the .s member.
+#define OP_OSTREAM(Ty, VecSize)                                                \
+    std::ostream& operator<<(std::ostream& os, const Ty##VecSize& val)         \
+    {                                                                          \
+        os << +val.s[0];                                                       \
```

Comment:
I'm probably missing something, but why do we need the `+` before `val.s[x]`?

Reply:
It's to ensure `char`s get printed as a number instead of being rendered as a character.  Perhaps I should add a comment to explain that.

Reply:
ah that makes sense, a quick comment would be great

Reply:
Added a comment.